From 30f5d53244ef4173544413124f319af272235cfd Mon Sep 17 00:00:00 2001
From: Madhurkiran Harikrishnan <madhurki@xilinx.com>
Date: Wed, 16 May 2018 17:16:53 -0700
Subject: [PATCH 1/3] qwaylandwindow.cpp: Do not destroy shell suface before
 eglDestroySurface

The call to reset results in a race condition resulting in destruction
of shell surface before the eglSurface thereby crashing libwayland-client.

Signed-off-by: Madhurkiran Harikrishnan <madhurki@xilinx.com>
Upstream-Status: Pending
---
 src/client/qwaylandwindow.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/client/qwaylandwindow.cpp b/src/client/qwaylandwindow.cpp
index 6d7c088..94c89a3 100644
--- a/src/client/qwaylandwindow.cpp
+++ b/src/client/qwaylandwindow.cpp
@@ -385,8 +385,6 @@ void QWaylandWindow::setVisible(bool visible)
         QWindowSystemInterface::flushWindowSystemEvents();
         if (!deleteGuard.isNull() && window()->type() == Qt::Popup)
             closePopups(this);
-        if (!deleteGuard.isNull())
-            reset();
     }
 }
 
-- 
2.7.4

